recommendation/mozlog/middleware.py

```diff
@@ -42,7 +42,10 @@ def request_summary(response):
 
     log = {}
     query = request.args.get('q')
-    data = response.get_data(as_text=True)
+    try:
+        data = response.get_data(as_text=True)
+    except UnicodeDecodeError:
+        data = ''
```

Comment:
The body is only used if we also get a query. What about moving this block into the ``if query:`` block later on, so it doesn't get called for the new image API at all?